homeassistant/components/hvac/__init__.py

```diff
@@ -0,0 +1,492 @@
+"""
+Provides functionality to interact with hvacs.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/hvac/
+"""
+import logging
+import os
+
+from homeassistant.helpers.entity_component import EntityComponent
+
+from homeassistant.config import load_yaml_config_file
+import homeassistant.util as util
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.temperature import convert
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA  # noqa
+from homeassistant.components import zwave
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_TEMPERATURE, STATE_ON, STATE_OFF, STATE_UNKNOWN,
+    TEMP_CELCIUS)
+
+DOMAIN = "hvac"
+
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+SCAN_INTERVAL = 60
+
+SERVICE_SET_AWAY_MODE = "set_away_mode"
+SERVICE_SET_AUX_HEAT = "set_aux_heat"
+SERVICE_SET_TEMPERATURE = "set_temperature"
+SERVICE_SET_FAN_MODE = "set_fan_mode"
+SERVICE_SET_OPERATION_MODE = "set_operation_mode"
+SERVICE_SET_SWING = "set_swing_mode"
+SERVICE_SET_HUMIDITY = "set_humidity"
+
+STATE_HEAT = "heat"
+STATE_COOL = "cool"
+STATE_IDLE = "idle"
+STATE_AUTO = "auto"
+STATE_DRY = "dry"
+STATE_FAN_ONLY = "fan_only"
+
+ATTR_CURRENT_TEMPERATURE = "current_temperature"
+ATTR_CURRENT_HUMIDITY = "current_humidity"
+ATTR_HUMIDITY = "humidity"
+ATTR_AWAY_MODE = "away_mode"
+ATTR_AUX_HEAT = "aux_heat"
+ATTR_FAN_MODE = "fan_mode"
+ATTR_FAN_LIST = "fan_list"
+ATTR_MAX_TEMP = "max_temp"
+ATTR_MIN_TEMP = "min_temp"
+ATTR_MAX_HUMIDITY = "max_humidity"
+ATTR_MIN_HUMIDITY = "min_humidity"
+ATTR_OPERATION = "operation_mode"
+ATTR_OPERATION_LIST = "operation_list"
+ATTR_SWING_MODE = "swing_mode"
+ATTR_SWING_LIST = "swing_list"
+
+_LOGGER = logging.getLogger(__name__)
+
+DISCOVERY_PLATFORMS = {
+    zwave.DISCOVER_HVAC: 'zwave'
+}
+
+
+def set_away_mode(hass, away_mode, entity_id=None):
```

Comment:
A demo platform is a requirement for the introduction of a new component.
How else will you be able to write tests for all the different services.

On Tue, Apr 26, 2016, 15:35 John Arild Berentsen <notifications@github.com>
wrote:

> In homeassistant/components/hvac/__init__.py
> <https://github.com/home-assistant/home-assistant/pull/1912#discussion_r61085553>
> :
>
> > +ATTR_MIN_TEMP = "min_temp"
> > +ATTR_MAX_HUMIDITY = "max_humidity"
> > +ATTR_MIN_HUMIDITY = "min_humidity"
> > +ATTR_OPERATION = "operation_mode"
> > +ATTR_OPERATION_LIST = "operation_list"
> > +ATTR_SWING_MODE = "swing_mode"
> > +ATTR_SWING_LIST = "swing_list"
> > +
> > +_LOGGER = logging.getLogger(__name__)
> > +
> > +DISCOVERY_PLATFORMS = {
> > +    zwave.DISCOVER_HVAC: 'zwave'
> > +}
> > +
> > +
> > +def set_away_mode(hass, away_mode, entity_id=None):
>
> First I want to get the code of the component up to standards, then I
> start on making the Demo. I actually was thinking of making a configurable
> hvac platform, so a user can select by config a fan switch, heater switch
> etc. Like the heat_control platform.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/1912/files/403c27395c6cc511c04f1b241a05b95da265f2b0#r61085553>
>